File: dotnet/Allmark/Utils/AddMarkupAsText.cs
namespace Allmark;

using Allmark.Types;

public static partial class Utils
{
    public static void AddMarkupAsText(string markup, InlineParserState state, MarkdownNode parent)
    {
        var lastNode = parent.Children!.LastOrDefault();
        var haveText = lastNode != null && lastNode.Type == "text";
        var text = haveText ? lastNode! : NewText(state.I, state.Line, "", 0);
        text.Content += markup;
        if (!haveText)
        {
            parent.Children ??= new List<MarkdownNode>();
            parent.Children.Add(text);
        }
        state.I += markup.Length;
    }
}

File: swift/Sources/Allmark/Utils/AddMarkupAsText.swift
import Foundation

func addMarkupAsText(
	markup: String,
	state: inout InlineParserState,
	parent: inout MarkdownNode
) {
	let lastNode = parent.children?.last
	let haveText = lastNode?.type == "text"
	let text = haveText ? lastNode! : newInline(
		type: "text",
		index: state.i,
		line: state.line,
		markup: "",
		indent: 0
	)
	text.markup += markup
	if !haveText {
		parent.children?.append(text)
	}
	state.i += markup.count
}
